dali/dfu/dfurun.cpp

```diff
@@ -1614,7 +1613,7 @@ class CDFUengine: public CInterface, implements IDFUengine
                     fsys.replicate(fdesc.get(), mode, recovery, recoveryconn, filter, opttree, &feedback, &abortnotify, dfuwuid);
                     runningconn.clear();
                     if (!abortnotify.abortRequested()) {
-                        Audit("REPLICATE",userdesc,srcFile?srcFile->queryLogicalName():NULL,NULL);
```

Comment:
line 1611: Owned<IFileDescriptor> fdesc = srcFile->getFileDescriptor();

if srcFile is valid - so is srcFDec, so you can replace this.

Reply:
line 1580:                        srcFile.clear();

Now you have introduced a srcFDesc at same scope, you should clear it here as well.